wgsl/index.bs

```diff
@@ -6607,14 +6803,25 @@ When an identifier is used as a [=syntax/callable=] item, it is one of:
 
     | [=syntax/ident=]
 
-    | [=syntax/callable=] [=syntax/argument_expression_list=]
+    | [=syntax/attribute=] * [=syntax/call_expr=]
 
     | [=syntax/literal=]
 
     | [=syntax/paren_expression=]
 
     | [=syntax/bitcast=] [=syntax/less_than=] [=syntax/type_specifier=] [=syntax/greater_than=] [=syntax/paren_expression=]
 </div>
+<div class='syntax' noexport='true'>
+  <dfn for=syntax>call_expr</dfn> :
+
+    | [=syntax/call_phrase=]
+</div>
+Note: The [=syntax/call_expr=] rule exists to ensure [=type checking=] applies to the call expression.
```

Comment:
Can't type checking apply to call_phrase (or call_expr becoming the name) directly?

Reply:
Hm. I should have made it 'call_expression' instead.

I was thinking about this rule at the top of the type checking section:

> An expression is a segment of source text parsed as one of the WGSL grammar rules whose name ends with "_expression". 

Nothing says a 'call_phrase'  has to be type checked.  

The difficulty is when functions don't return anything. They don't have a result type.  I thought the split between call_phrase and call_expression would be needed for that.  I'm interested in what you think.

Reply:
@dneto0 I think because the Type Checking section says "values that may result from evaluating a particular expression" (with "may") after introducing contexts, it might be possible to accommodate without nesting rules? Maybe we can think of how functions that do not return a value in the callable choice of primary_expression bounds the concern and apply it here.

Reply:
The "may" in that sentence is about whether the expression is evaluated or not.  It might not be evaluated due to control flow.

Later the spec says:

> Type checking a successfully parsed WGSL program is the process of mapping each expression to its static type, and verifying that type requirements of each statement are satisfied.

That is unconditional. It must be done for every expression in the program.  And therefore the text matching the call_expression rule.

That was my thinking on this.

Reply:
Would `| attribute * call_phrase` without nesting not work (since it attaches to primary_expression)?